corehq/apps/app_manager/util.py

```diff
@@ -638,6 +639,32 @@ def get_latest_enabled_build_for_profile(domain, profile_id):
         return get_app(domain, latest_enabled_build.build_id)
 
 
+@quickcache(['domain', 'location_id', 'app_id'], timeout=24 * 60 * 60)
+def get_latest_enabled_app_release(domain, location_id, app_id):
+    """
+    for a location search for enabled app releases for all parent locations.
+    Child location's setting takes precedence over parent
+    """
+    from corehq.apps.app_manager.models import LatestEnabledAppRelease
+    location = SQLLocation.active_objects.get(location_id=location_id)
+    location_and_ancestor_ids = location.get_ancestors(include_self=True).values_list(
+        'location_id', flat=True).reverse()
+    # get all active enabled releases and order by version desc to get one with the highest version in the end
+    # for a location. Do not use the first object itself in order to respect the location hierarchy and use
+    # the closest location to determine the valid active release
+    latest_enabled_releases = {
```

Comment:
if you are relying on dict ordering here, you should use an `OrderedDict` (until we switch to Python 3.7+)